### PR TITLE
[FREELDR] Remove a few useless code lines

### PR DESCRIPTION
**Pull request by SergeGautherie**
Reduce FreeLdr memory size.

Also reported by CppCheck:
```
BytesToRead -= BytesToRead;<--- Same expression on both sides of '-='.

    if (Data != NULL)<--- Condition 'Data!=NULL' is always true
```

JIRA issue: [CORE-14516](https://jira.reactos.org/browse/CORE-14516)

### CODE REVIEW

File: boot/freeldr/freeldr/lib/fs/fat.c [hunk 1]
```diff
@@ -1352,8 +1352,6 @@ BOOLEAN FatReadFile(PFAT_FILE_INFO FatFileInfo, ULONG BytesToRead, ULONG* BytesR
             *BytesRead += BytesToRead;
         }
         FatFileInfo->FilePointer += BytesToRead;
-        BytesToRead -= BytesToRead;
-        Buffer = (PVOID)((ULONG_PTR)Buffer + BytesToRead);
```
Comment (HBelusca) [inline boot/freeldr/freeldr/lib/fs/fat.c:1356]: You can also fix the associated comment above, into
```
// Now do the read and update BytesRead & FilePointer
```

File: boot/freeldr/freeldr/lib/fs/ext2.c [hunk 2]
```diff
@@ -533,8 +533,6 @@ BOOLEAN Ext2ReadFileBig(PEXT2_FILE_INFO Ext2FileInfo, ULONGLONG BytesToRead, ULO
             *BytesRead += BytesToRead;
         }
         Ext2FileInfo->FilePointer += BytesToRead;
-        BytesToRead -= BytesToRead;
```
Comment (HBelusca) [inline boot/freeldr/freeldr/lib/fs/ext2.c:536]: Same comment as the one I have for fat.c

**Review by HBelusca — COMMENTED** [review]
Minor comments, but LGTM.

File: boot/freeldr/freeldr/lib/fs/ext2.c (hunk 2 above)
Comment (SergeGautherie) [reply boot/freeldr/freeldr/lib/fs/ext2.c:536]: Obviously.

File: boot/freeldr/freeldr/lib/fs/fat.c (hunk 1 above)
Comment (SergeGautherie) [reply boot/freeldr/freeldr/lib/fs/fat.c:1356]: Right. Updated.

**Pull request merged**